src/altinn-app-frontend/src/components/base/ListComponent.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+
+import {
+  Pagination,
+  SortDirection,
+  Table,
+  TableBody,
+  TableCell,
+  TableFooter,
+  TableHeader,
+  TableRow,
+} from '@altinn/altinn-design-system';
+import type * as altinnDesignSystem from '@altinn/altinn-design-system';
+
+import type { PropsFromGenericComponent } from '..';
+
+import { useAppDispatch, useAppSelector, useHasChangedIgnoreUndefined } from 'src/common/hooks';
+import { useGetDataList } from 'src/components/hooks';
+import { useDelayedSavedState } from 'src/components/hooks/useDelayedSavedState';
+import { dataListsActions } from 'src/shared/resources/dataLists/dataListsSlice';
+
+export type IListProps = PropsFromGenericComponent<'List'>;
+
+const defaultDataList: any[] = [];
+
+export const ListComponent = ({
+  tableHeaders,
+  fieldToStoreInDataModel,
+  dataList,
+  dataListId,
+  mapping,
+  pagination,
+  formData,
+  handleDataChange,
+  sortableColumns,
+}: IListProps) => {
+  const dynamicDataList = useGetDataList({ dataListId, mapping });
+  const calculatedDataList = dynamicDataList || defaultDataList;
+
+  const rowsPerPage = useAppSelector(
+    (state) => state.dataListState.dataLists[dataListId || ''].size || pagination.default,
+  );
+  const currentPage = useAppSelector((state) => state.dataListState.dataLists[dataListId || ''].pageNumber || 0);
```

Comment:
Have you tested this with multiple `List` components each using the same `dataListId` on the same page? I suspect they will be linked, but it should be possible to use multiple components mapped to the same `dataListId` individually. My suggestion would be to use the `id` property passed to the component as the key in the redux state instead of `dataListId`.